container/Exchange.js

```diff
@@ -28,19 +44,13 @@ export default () => {
 
     return () => {
       cancelled = true
+      setTranslation(undefined)
     }
   }, [search])
 
   return (
     <Container>
-      <h1>Exchange</h1>
-      <div>
-        <Bold>Loading:</Bold> {loading.toString()}
-      </div>
-      <p>
-        <Bold>Result:</Bold>
-      </p>
-      <div>{JSON.stringify(translation)}</div>
+      {translation ? <Translation translationResult={translation} /> : loading}
```

Comment:
This was the reason why I called `setTranslation(undefined)` to clear the old translation for the loading thing; I have to check first that there is actually a word being searched since 'Translation' takes in '{word, translation}'. Right now it doesn't exist at all, but if I call 'setTranslation(undefined)' then at least `translation` equates to something. When I check if it's just loading, I get the same TypeError: [https://imgur.com/a/lOPuFCJ](url)

Reply:
I think you should check _both_ if a translation exists, as well as if it is loading.

```javascript
(loading || !translation)
  ? <Spinner />
  : <Translation translationResult={translation} />
```

Also, it might be a separate bug that an empty word results in a search.

Reply:
Maybe `<Translation />` should be able to see if there is a result but no found translations. In a later PR.

![Screen Shot 2020-03-02 at 09 22 51](https://user-images.githubusercontent.com/8337202/75657884-7c98d880-5c67-11ea-9d2a-dba827892713.png)